decorators/ng-module/metadata-handlers/provider.ts

```diff
@@ -0,0 +1,21 @@
+export default function providerHandler(ng1Module: any, providers: any, declarations: any) {
+    declarations.forEach((declaration: any) => {
+        const injections = Reflect.getMetadata("design:paramtypes", declaration);
+
+        if (injections) {
+            const injectedServices = injections.map(({ name }: any) => name);
+
+            if (!declaration.$inject) {
+                declaration.$inject = [];
+            }
+
+            providers.forEach((provider: any) => {
+                const serviceToken = provider.name;
+                const injectIndex = injectedServices.indexOf(serviceToken);
+
+                declaration.$inject[injectIndex] = serviceToken;
```

Comment:
We can't, because the array of injections will be created every time when new loop starts.